src/init.cpp

```diff
@@ -632,6 +633,11 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     // ********************************************************* Step 0: Load zcash params
     ZC_LoadParams();
 
+    // Perform libsodium initialization
+    if (sodium_init() == -1) {
```

Comment:
It's not clear to me just based on the names of functions whether or not ``sodium_init`` needs to happen before ``ZC_LoadParams``. Since ``libsodium`` is a dependency of the zerocash stuff, perhaps it should always be initialized prior to any "libzerocash" level stuff.